addons/giltracker/giltracker.lua

```diff
@@ -0,0 +1,212 @@
+        LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+        ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+        (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+        SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+_addon.name = 'giltracker'
+_addon.author = 'sylandro'
+_addon.version = '1.0.0'
+_addon.language = 'English'
+
+config = require('config')
+images = require('images')
+texts = require('texts')
+
+local GIL_ITEM_ID = 0xFFFF
+local CUTSCENE_STATUS_ID = 4
+local SCROLL_LOCK_KEY = 70
+
+hideKey = SCROLL_LOCK_KEY
+is_hidden_by_cutscene = false
+is_hidden_by_key = false
+
+defaults = {}
+defaults.hideKey = SCROLL_LOCK_KEY
+defaults.gilText = {}
+defaults.gilText.bg = {}
+defaults.gilText.bg.alpha = 100
+defaults.gilText.bg.red = 0
+defaults.gilText.bg.green = 0
+defaults.gilText.bg.blue = 0
+defaults.gilText.bg.visible = false
+defaults.gilText.text = {}
+defaults.gilText.text.font = 'sans-serif'
+defaults.gilText.text.fonts = {'Arial','Trebuchet MS'}
+defaults.gilText.text.size = 9
+defaults.gilText.flags = {}
+defaults.gilText.flags.italic = true
+defaults.gilText.flags.bold = false
+defaults.gilText.flags.right = true
+defaults.gilText.flags.bottom = true
+defaults.gilText.pos = {}
+defaults.gilText.pos.x = -285
+defaults.gilText.pos.y = -35
+defaults.gilText.text.alpha = 255
+defaults.gilText.text.red = 253
+defaults.gilText.text.green = 252
+defaults.gilText.text.blue = 250
+defaults.gilText.text.stroke = {}
+defaults.gilText.text.stroke.alpha = 200
+defaults.gilText.text.stroke.red = 50
+defaults.gilText.text.stroke.green = 50
+defaults.gilText.text.stroke.blue = 50
+defaults.gilText.text.stroke.width = 2
+defaults.gilText.text.visible = true
+defaults.gilImage = {}
+defaults.gilImage.color = {}
+defaults.gilImage.color.alpha = 255
+defaults.gilImage.color.red = 255
+defaults.gilImage.color.green = 255
+defaults.gilImage.color.blue = 255
+defaults.gilImage.visible = true
+
+local settings = config.load(defaults)
+config.save(settings)
+
+settings.gilImage.texture = {}
+settings.gilImage.texture.path = windower.addon_path..'gil.png'
+settings.gilImage.texture.fit = true
+settings.gilImage.size = {}
+settings.gilImage.size.height = 23
+settings.gilImage.size.width = 23
+settings.gilImage.draggable = false
+settings.gilImage.repeatable = {}
+settings.gilImage.repeatable.x = 1
+settings.gilImage.repeatable.y = 1
+
+gil_image = images.new(settings.gilImage)
+gil_text = texts.new(settings.gilText)
+
+config.register(settings, function(settings)
+    hideKey = settings.hideKey
+end)
+
+windower.register_event('load',function()
+    if windower.ffxi.get_info().logged_in then
+        initialize()
+    end
+end)
+
+windower.register_event('login', function()
+    initialize()
+end)
+
+windower.register_event('logout', function(...)
+    hide()
+end)
+
+windower.register_event('add item', function(...)
```

Comment:
The function signature needs to be adjusted:
```lua
windower.register_event(function(bag, slot, id, count)
    update_gil_if_item_id_matches(id)
end)
```

Otherwise the `id` parameter will always be `nil` and the check in the following function will always fail.

And ftr, there's something some people do in Lua, if a parameter is unused (since this function has four parameters, but you only use one of them), which is to name all other parameters `_`, to indicate that they are unused. Don't have to do that, some people also prefer not to do it to keep the signature accurate. Just letting you know.

Reply:
```lua
windower.register_event('add item', function(bag, slot, id, count)
    update_gil_if_item_id_matches(id)
end)
```

:D